resources/analysisTools/qcPipeline/workflowLib.sh

```diff
@@ -57,7 +57,11 @@ getBigScratchDirectory () {
     echo "$scratchDir"
 }
 
-
+getFilePathOnRoddyScratch () {
+    local sourceLocation="${1}"
+    basename=$(basename ${sourceLocation})
+    echo "${RODDY_SCRATCH}/${basename}"
```

Comment:
1. No argument should result in error.
2. `basename` must be local or will be global.
3. Don't use `${...}` unless necessary (e.g. if default value, error messages, or any other Bash feature that requires that syntax. The `{}` clutter the picture)
```suggestion
    local sourceLocation="${1:?No source location}"
    local basename="$(basename "$sourceLocation")"
    echo "$RODDY_SCRATCH/$basename"
```